src/context/client.rs

```diff
@@ -19,48 +20,92 @@ impl RedisModuleClientInfo {
 }
 
 /// GetClientNameById, GetClientUserNameById and GetClientCertificate use autoMemoryAdd on the ValkeyModuleString pointer
-// after the callback (command, server event handler, ...) these ValkeyModuleString pointers will be freed automatically
+/// after the callback (command, server event handler, ...) these ValkeyModuleString pointers will be freed automatically
 impl Context {
     pub fn get_client_id(&self) -> u64 {
         unsafe { RedisModule_GetClientId.unwrap()(self.ctx) }
     }
 
-    pub fn get_client_name(&self) -> ValkeyString {
-        let client_id = self.get_client_id();
+    /// wrapper for RedisModule_GetClientNameById
+    pub fn get_client_name_by_id(&self, client_id: u64) -> ValkeyString {
         let client_name = unsafe { RedisModule_GetClientNameById.unwrap()(self.ctx, client_id) };
         ValkeyString::from_redis_module_string(self.ctx, client_name)
     }
 
-    pub fn set_client_name(&self, client_name: &ValkeyString) -> i64 {
-        let client_id = self.get_client_id();
+    /// wrapper for RedisModule_GetClientNameById using current client ID
+    pub fn get_client_name(&self) -> ValkeyString {
+        self.get_client_name_by_id(self.get_client_id())
+    }
+
+    /// wrapper for RedisModule_SetClientNameById
+    pub fn set_client_name_by_id(&self, client_id: u64, client_name: &ValkeyString) -> i64 {
         let resp = unsafe { RedisModule_SetClientNameById.unwrap()(client_id, client_name.inner) };
```

Comment:
There are a few error cases that are possible with this API:
* The client of provided ID does not exist 
* The client name is invalid

We are returning a code back here which is good. But, we could return `Status` for clearer interface for Modules to build against

https://docs.rs/valkey-module/latest/valkey_module/raw/enum.Status.html#variant.Ok

Reply:
currently set_client_name_by_id with invalid ID will return 1 and with valid ID will return 0.  I can make it Status.